src/core/installAddon.js

```diff
@@ -72,15 +72,32 @@ export function makeProgressHandler(dispatch, guid) {
 }
 
 export function mapStateToProps(state, ownProps) {
+  const guid = ownProps.guid || (ownProps.addon && ownProps.addon.guid);
```

Comment:
As discussed on vidyo this will resolve to a guid on the addon object if a guid directly on ownProps does not exist. Since this is now used in multiple places it should be factored out to it's own function.